src/test/java/pages/SongsPage.java

```diff
@@ -0,0 +1,88 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import java.time.Duration;
+import java.util.List;
+
+public class SongsPage extends BasePage{
+
+    public SongsPage(WebDriver givenDriver) {
+        super(givenDriver);
+    }
+
+    @FindBy(css = "#songsWrapper > div > div > div.item-container > table")
+        WebElement songTable;
+    @FindBy(css = "#songResultsWrapper > div > div > div.item-container > table")
+        WebElement searchResultsTable;
+    @FindBy(xpath = "//button[@data-test='view-all-songs-btn']")
+        WebElement buttonViewAll;
+    @FindBy(css = "#songResultsWrapper > header > div.song-list-controls > span > button.btn-add-to")
+        WebElement clickAddTo;
```

Comment:
Hi,
I have checked your homework 23, you have implemented Page Factory, used @findby annotation for declaring Web Elements. Great! Keep up the excellent work!
But I have some suggestion for you:
- When define locator for element, best pratices for locators:
```
1. Start with the folling order id, unique attribute, css, xpath
2. Never use Absoulute paths, always use relative ones;
3. Your locator should be unique
4. Keep your locator simple and robust
5. Don't locate elements by visual labels (not reliable, may change often)
```
You are defining a locator for element so long and using Absoulute paths
//Example:
```
@FindBy(css = "#songResultsWrapper button.btn-add-to")
WebElement clickAddTo;
```